### PR TITLE
add os.getCurrentProcessId()

### PR DESCRIPTION
**Pull request by timotheecour**
/cc @Araq 
surprising it didn't exist yet

after this is merged maybe nimble/src/nimblepkg/tools.nim can be refactored use that  /cc @dom96 

## links
* [How do I get the process ID of the current program? - Nim forum](https://forum.nim-lang.org/t/946)

### CODE REVIEW

File: lib/pure/osproc.nim [hunk 1]
```diff
@@ -158,6 +158,15 @@ proc processID*(p: Process): int {.rtl, extern: "nosp$1".} =
   ## returns `p`'s process ID.
   return p.id
 
+proc processID*(): int =
+  ## return current process ID
+  when defined(windows):
+    proc GetCurrentProcessId(): int32 {.stdcall, dynlib: "kernel32",
```
Comment (data-man) [inline lib/pure/osproc.nim]: int32?
DWORD is unsigned int in Windows headers.

Comment (timotheecour) [reply lib/pure/osproc.nim]: fixed, thanks! btw, you can use the new github feature "insert a suggestion" so PR author can just click "accept suggestion"

File: lib/pure/osproc.nim [hunk 2]
```diff
@@ -158,6 +158,16 @@ proc processID*(p: Process): int {.rtl, extern: "nosp$1".} =
   ## returns `p`'s process ID.
   return p.id
 
+proc processID*(): int =
+  ## return current process ID
+  when defined(windows):
+    type DWORD = uint32
+    proc GetCurrentProcessId(): DWORD {.stdcall, dynlib: "kernel32",
```
Comment (data-man) [inline lib/pure/osproc.nim]:
```suggestion
    proc GetCurrentProcessId(): uint {.stdcall, dynlib: "kernel32",
```

Comment (data-man) [reply lib/pure/osproc.nim]: uint32 in 32 bit target
uint64 in 64 bit target

Comment (timotheecour) [reply lib/pure/osproc.nim]: are you sure? 

see
* https://stackoverflow.com/questions/39419/how-large-is-a-dword-with-32-and-64-bit-code
* https://msdn.microsoft.com/en-us/library/cc230318.aspx

Comment (data-man) [reply lib/pure/osproc.nim]: I'm not sure. :-)
Hmm `DWORD* = int32` in winlean.nim
Why?!

**Comment by dom96** [comment]
Hehe: https://github.com/nim-lang/nimble/blob/3ee1e115f4aa250706baf1da2c562f5326f1c98d/src/nimblepkg/tools.nim#L159

**Comment by timotheecour** [comment]
> Hehe: https://github.com/nim-lang/nimble/blob/3ee1e115f4aa250706baf1da2c562f5326f1c98d/src/nimblepkg/tools.nim#L159

@dom96 I know, see https://github.com/nim-lang/nimble/issues/591

> This should go into os IMO

you're probably right given it uses nothing in `osproc`; only reason i had put it there was there was already a 
`proc processID*(p: Process): int {.rtl, extern: "nosp$1".} =` there; but I guess we can re-export `processID` from osproc

**Comment by timotheecour** [comment]
@dom96 done, PTAL

File: lib/pure/osproc.nim (hunk 2 above)
Comment (timotheecour) [reply lib/pure/osproc.nim]: good catch, filed https://github.com/nim-lang/Nim/issues/9848

**Comment by Araq** [comment]
changelog entry would be nice too.

**Comment by timotheecour** [comment]
> changelog entry would be nice too.

done (and oups, forgot to add [ci-skip], but it was green before changelog entry addition)

File: lib/pure/osproc.nim [hunk 3]
```diff
@@ -15,7 +15,7 @@ include "system/inclrtl"
 import
   strutils, os, strtabs, streams, cpuinfo
 
-export quoteShell, quoteShellWindows, quoteShellPosix
+export quoteShell, quoteShellWindows, quoteShellPosix, processID
```
Comment (Araq) [inline lib/pure/osproc.nim]: No, there should be one way to import this symbol and it is in ``os.nim``.

Comment (timotheecour) [reply lib/pure/osproc.nim]: done; (although it's quite odd that we have these 2 overloads split up: `os.processID()` and `osproc.processID(Process)`)

**Comment by timotheecour** [comment]
PTAL

**Comment by dom96** [comment]
Actually. I would rename it to getCurrentProcessId in os.nim

**Comment by timotheecour** [comment]
> Actually. I would rename it to getCurrentProcessId in os.nim

how about `getCurrentPid()` ? such a common name, it sucks if it's as long as `getCurrentProcessId`
we already have `getCurrentDir`, not `getCurrentDirectory`

**Comment by Araq** [comment]
Long names don't suck for rarely used stuff. And its name shall be ``getCurrentProcessId``

**Comment by timotheecour** [comment]
PTAL

**Pull request merged**